app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onBitmapDrawn: ((Bitmap) -> Unit)? = null
+    var bitmap: Bitmap? = null
+    private var shape: Rect? = null
+
+    var pathColor = Color.BLACK
+    private var pathPaint = Paint().apply {
+        color = pathColor
+        style = Paint.Style.STROKE
+        strokeWidth = 20f
+        strokeJoin = Paint.Join.ROUND
+        strokeCap = Paint.Cap.ROUND
+    }
+
+    private var bitmapPaint = Paint()
+    lateinit var path: Path
```

Comment:
Does this really need to be lateinit? Can null be better?

Reply:
made nullable